### DISCUSSION TITLE
Capability of ortools supporting the 3D bin packing problem and visualization

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#2741] [Feature requests] [answered]
Hi there, 

I am wondering if ortools have applications and examples about the 3D bin packing problem? I am facing a problem of planning the loading of thousands of items/boxes/pallets into the ocean containers (mainly 40HC). 

Mannual planning, as most companies currently are doing, is very inefficient and painful. I am very interested to know if there is any ortool-based or other optimization toolkit that helps to solve such problem in a relatively accurate manner.

Allocating the items to containers are relatively easy, as I see from the example: https://developers.google.com/optimization/bin/bin_packing#complete_programs 
What is challenging is to generate a concrete plan telling people how to load each items into a specific container, like which box needs to be placed inside first, which item needs to be placed on top of others, so that the container storage space is maximally utilized.
And I hope to have a visual ouput like below:
![360截图20210827202414341](https://user-images.githubusercontent.com/34776946/131128644-d05d280b-43c1-42a8-9d2b-885898fa4ffa.jpg)
![360截图20210827202111136](https://user-images.githubusercontent.com/34776946/131128672-1c71230c-1653-4655-82ae-b2fb506fb333.jpg)

**Comment by Mizux** [comment, marked as answer]
https://github.com/MrBenGriffin/or-tools-fun#polycube-puzzle-solver ?
